syntax = "proto3";
package msg;

import "google/protobuf/timestamp.proto";


service MetricService {
    rpc SendMetrics(Metric) returns (MetricsReply) {}
    rpc UploadAsset(stream AssetUpload) returns (AssetUploadReply) {}
}


message Metric {
    string experimentID = 1;
    string metricBody = 2;
    int64 createdAt = 3;

}

message MetricsReply {
    int32 statusCode = 1;
    string message = 2;
}

message AssetUpload {
    oneof data {
        AssetInfo info = 1;
        bytes content = 2;
    }
}

message AssetInfo {
    string AssetName = 1 ;
    string AssetType = 2;
}

message AssetUploadReply {
    string message = 1;
}